File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/repository/UserRepository.java
package com.lattis.ellipse.domain.repository;

import com.lattis.ellipse.data.network.model.response.AddPrivateNetworkResponse;
import com.lattis.ellipse.data.network.model.response.BasicResponse;
import com.lattis.ellipse.data.network.model.response.GetCurrentUserStatusResponse;
import com.lattis.ellipse.domain.model.TermsAndConditions;
import com.lattis.ellipse.domain.model.User;

import io.reactivex.Observable;


public interface UserRepository {

    Observable<User> getUser();

    Observable<User> saveUser(User user);

    Observable<User> saveUserLocally(User user);

    Observable<Boolean> validateCodeForChangePhoneNumber(String code, String phoneNumber);

    Observable<Boolean> changePassword(String password, String new_password);

    Observable<Boolean> sendCodeToUpdatePhoneNumber(String countryCode, String phoneNumber);

    Observable<Boolean> sendCodeToUpdateEmail(String email);

    Observable<GetCurrentUserStatusResponse> getCurrentUserStatus();

    Observable<TermsAndConditions> getTermsAndConditions();

    Observable<Boolean> acceptTermsAndCondition(boolean accepted);

    Observable<Boolean> checkTermsAndConditionAccepted();

    Observable<BasicResponse> sendForgotPasswordCode(String email);

    Observable<BasicResponse> confirmCodeForForgotPassword(String email, String code, String password);

    Observable<Boolean> validateCodeForChangeEmail(String code, String email);
    Observable<Boolean> confirmVerificationCodeForPrivateNetwork(String userId, String account_type, String confirmationCode);


    Observable<AddPrivateNetworkResponse> addPrivateNetworkEmail(String email);

    Observable<Boolean> deleteUserAccount();

    Observable<User> getLocalUser();
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/repository/UserRepository.kt
package com.lattis.domain.repository

import com.lattis.domain.models.Bike
import com.lattis.domain.models.UserCurrentStatus
import com.lattis.domain.models.PrivateNetwork
import com.lattis.domain.models.User
import io.reactivex.rxjava3.core.Observable

interface UserRepository{

    fun getUserCurrentStatus(): Observable<UserCurrentStatus>
    fun getUser(): Observable<User>
    fun saveUserLocally(user: User): Observable<User>
    fun saveUser(user: User): Observable<User>
    fun getLocalUser():Observable<User>
    fun confirmVerificationCodeForPrivateNetwork(
        userId: String,
        account_type: String,
        confirmationCode: String
    ): Observable<Boolean>
    fun addPrivateNetworkEmail(email: String): Observable<List<PrivateNetwork>>

    fun sendCodeToUpdatePhoneNumber(
        countryCode: String,
        phoneNumber: String
    ): Observable<Boolean>

    fun validateCodeForChangePhoneNumber(
        code: String,
        phoneNumber: String
    ): Observable<Boolean>

    fun changePassword(
        password: String,
        new_password: String
    ): Observable<Boolean>

    fun validateCodeForEmailChange(
        code: String,
        email: String
    ): Observable<Boolean>

    fun sendCodeToUpdateEmail(email: String): Observable<Boolean>

    fun sendForgotPasswordCode(email: String): Observable<Boolean>

    fun confirmCodeForForgotPassword(
        email: String,
        code: String,
        password: String
    ): Observable<Boolean>

    fun getUserFleet():Observable<List<Bike.Fleet>>
    fun deleteAccount():Observable<Boolean>
}